trie/stacktrie.go

```diff
@@ -202,39 +119,20 @@ const (
 	hashedNode
 )
 
-// Update inserts a (key, value) pair into the stack trie.
-func (st *StackTrie) Update(key, value []byte) error {
-	k := keybytesToHex(key)
-	if len(value) == 0 {
-		panic("deletion not supported")
-	}
-	st.insert(k[:len(k)-1], value, nil)
-	return nil
-}
-
-// MustUpdate is a wrapper of Update and will omit any encountered error but
-// just print out an error message.
-func (st *StackTrie) MustUpdate(key, value []byte) {
-	if err := st.Update(key, value); err != nil {
-		log.Error("Unhandled trie error in StackTrie.Update", "err", err)
-	}
-}
-
-func (st *StackTrie) Reset() {
-	st.owner = common.Hash{}
-	st.writeFn = nil
+func (st *stNode) Reset() *stNode {
 	st.key = st.key[:0]
 	st.val = nil
 	for i := range st.children {
 		st.children[i] = nil
```

Comment:
I spent a long time stuck on this. 
Since we just set the children to nil, we could loose a lot of *stNodes that are not in the pool. However I finally came to the conclusion that this is not an issue since we only ever return *stNodes to the pool on
- hashRec of extNodes, which have no children themselves (extNode)
- hashRec of branchNodes, where we hashRec'd the children beforehand, thus returned the *stNodes already

So we will not loose allocations here (afaict)

Reply:
Yeah, this is probably not even needed. I tested this: 
```
		if st.children[i] != nil {
			panic(fmt.Sprintf("Child was %T", st.children[i]))
		}
```
But none of the tests triggered it. So the loop could probably be removed, but OTOH I think it's sane to have it there, to prevent potential mem leaks. It kind of makes sense that the pool-handling is robust in itself, without trusting too much that the "other parts" is correct.